src/test/java/de/retest/recheck/persistence/xml/XmlTransformerTest.java

```diff
@@ -0,0 +1,46 @@
+package de.retest.recheck.persistence.xml;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import java.nio.file.Files;
+import java.nio.file.Path;
+import java.nio.file.Paths;
+
+import org.junit.jupiter.api.Test;
+
+import de.retest.recheck.persistence.xml.util.StdXmlClassesProvider;
+import de.retest.recheck.ui.descriptors.Element;
+import de.retest.recheck.ui.descriptors.SutState;
+
+class XmlTransformerTest {
+
+	@Test
+	void should_not_mix_data_from_multiple_xml_files() throws Exception {
+		final String base = "src/test/resources/de/retest/recheck/persistence/xml/";
+
+		final XmlTransformer cut = new XmlTransformer( StdXmlClassesProvider.getXmlDataClasses() );
```

Comment:
Line 155 in class XmlTransformer sets the StdXmlClasses, so it's duplicated. An empty constructor is an option.